src/lib/utils/getWidgetAttribute.ts

```diff
@@ -0,0 +1,37 @@
+import { InitOptions, Widget, WidgetRenderState } from '../../types';
+
+export function getWidgetAttribute(
+  widget: Widget,
+  initOptions: InitOptions
+): string {
+  try {
+    // assume the type to be the correct one, but throw a nice error if it isn't the case
+    type WidgetWithAttribute = Widget<{
+      renderState: WidgetRenderState<
+        Record<string, any>,
+        { attribute: string } | { attributes: string[] }
+      >;
+    }>;
+
+    const {
+      widgetParams,
+    } = (widget as WidgetWithAttribute).getWidgetRenderState(initOptions);
+
+    const attribute =
+      'attribute' in widgetParams
+        ? widgetParams.attribute
+        : widgetParams.attributes[0];
+
+    if (typeof attribute !== 'string') throw new Error();
+
+    return attribute;
+  } catch (e) {
+    throw new Error(
+      `Could not find the attribute of the widget:
+
+${JSON.stringify(widget)}
```

Comment:
for custom widgets without type this will serialise as `{}`, do we see an alternative?

Reply:
We could just throw `Could not find the attribute of the custom widget.`, wdyt?